src/pages/payments/refund/Refund.tsx

```diff
@@ -0,0 +1,213 @@
+      setErrors(undefined);
+      axios
+        .post(
+          endpoint('/api/v1/payments/refund?&email_receipt=:email', { email }),
+          values,
+          {
+            headers: defaultHeaders,
+          }
+        )
+        .then(() => {
+          toast.success(t('updated_payment'), { id: toastId });
+        })
+        .catch((error: AxiosError) => {
+          console.error(error);
+          toast.error(t('error_title'), { id: toastId });
+          if (error.response?.status === 422) {
+            setErrors(error.response.data);
+          }
+        })
+        .finally(() => {
+          formik.setSubmitting(false);
+          queryClient.invalidateQueries(
+            generatePath(`/api/v1/payments/refund?&email_receipt=${email}`)
+          );
+        });
+    },
+  });
+
+  useEffect(() => {
+    invoices.map((invoiceId: string) => {
+      const invoiceItem = payment?.data.data.invoices.find(
+        (invoice: Invoice) => invoice.id == invoiceId
+      );
+      if (invoiceItem)
+        formik.setFieldValue('invoices', [
+          ...formik.values.invoices,
+          {
+            amount:
+              invoiceItem?.paid_to_date >
+              payment?.data.data.amount - payment?.data.data.refunded
+                ? payment?.data.data.amount - payment?.data.data.refunded
+                : invoiceItem?.paid_to_date,
+            invoice_id: invoiceItem?.id,
+            credit_id: '',
+            id: '',
+          },
+        ]);
+    });
+  }, [invoices]);
+
+  useEffect(() => {
+    let total = 0;
+    formik.values.invoices.map((invoice: any) => {
+      total = total + Number(invoice.amount);
+      setInvoices(
+        invoices.filter((invoiceId: string) => invoiceId != invoice.invoice_id)
+      );
+    });
+  }, [formik.values.invoices]);
+
+  return (
+    <Card
+      title={t('apply_payment')}
+      disableSubmitButton={formik.isSubmitting}
+      onFormSubmit={formik.handleSubmit}
+      withSaveButton
+    >
+      <Element leftSide={t('number')}>
+        <InputField disabled value={payment?.data.data.number} />
+      </Element>
+      <Element leftSide={t('amount')}>
+        <InputField
+          disabled
+          value={payment?.data.data.amount - payment?.data.data.refunded}
+        />
+      </Element>
+      <Element leftSide={t('applied')}>
+        <InputField disabled value={payment?.data.data.applied} />
+      </Element>
+      <Element leftSide={t('date')}>
+        <InputField
+          type="date"
+          value={formik.values.date}
+          onChange={formik.handleChange}
+        />
+      </Element>
+      <Element leftSide={t('invoices')}>
+        <SelectField
+          value=""
+          onChange={(event: any) => {
+            if (
+              formik.values.invoices.filter(
+                (invoices: any) => invoices.invoice_id == event.target.value
+              ).length < 1
+            )
+              setInvoices([...invoices, event.target.value]);
```

Comment:
1. Missing types (Invoice[])
2. What exactly are we filtering here?
3. Why do we have separate `invoices` state variable? Can you explain why it is needed?